backend/api/Controllers/RobotController.cs

```diff
@@ -277,6 +280,51 @@ public async Task<ActionResult<IList<VideoStream>>> GetVideoStreams([FromRoute]
         return Ok(robot.VideoStreams);
     }
 
+    /// <summary>
+    /// Get all plants associated with an active robot.
+    /// </summary>
+    /// <remarks>
+    /// <para> Retrieves the plants that have an active robot </para>
+    /// </remarks>
+    [HttpGet]
+    [Authorize(Roles = Role.User)]
+    [Route("active-plants")]
+    [ProducesResponseType(typeof(IList<EchoPlantInfo>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IList<EchoPlantInfo>>> GetActivePlants()
+    {
+        var plants = await _robotService.ReadAllActivePlants();
+
+        // TODO: call echo API here to get the remaining info so that we can rely solely on this endpoint
```

Comment:
Todo needs to be addressed, though as the comment above states, should this be swapped around?